src/commands/wallet.rs

```diff
@@ -27,185 +19,171 @@ pub(crate) enum Command {
         #[clap(long = "l1", required = false)]
         l1: bool,
     },
-    #[clap(about = "Deposit funds into the wallet.")]
+    #[clap(about = "Deposit funds into some wallet.")]
     Deposit {
-        #[clap(long = "amount", value_parser=U256::from_dec_str)]
-        amount: U256,
-        #[clap(long = "token")]
+        #[clap(long = "amount")]
+        amount: String,
+        #[clap(
+            long = "token",
+            help = "Specify the token address, the base token is used as default."
+        )]
```

Comment:
Why are we changing the amount's type?

Reply:
I see, then you use `parse_ether`. I think is ok, but I think it'd be better to keep the type and replace the `value_parser=U256::from_dec_str` with `value_parset=parse_ether`.

Reply:
The problem with this is that you then assume that every passed amount would be ETH. In this case, we could add a boolean flag `--eth` to handle this case and keep the original.

Reply:
Because it has to be parsed, it wasn't being parsed before, you had to know in advance that amount was in `wei`. Now if you want to send 5 tokens, you just pass 5 as argument and then the function `parse_ether` does the rest:
```rust
let parsed_amount = parse_ether(&amount)?;
```
`amount` has to be a String, that's the reason the type has to change.

Reply:
Yes, parse_ether works for any erc20 with 18 decimals, the problem is not assuming you are passing ETH, the problem is the token's decimals. There is another function that does this, but we will have to ask for the token decimals in advance.

Reply:
> value_parser=U256::from_dec_str with value_parser=parse_ether.

I've tried this, but since `parse_ether` is a function it doesn't work straigh taway.

Reply:
I think i found a workaround, i've just pushed